android/quest/src/test/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivityTest.kt

```diff
@@ -197,6 +205,67 @@ class QuestionnaireActivityTest : RobolectricTest() {
     Assert.assertNotNull(dialog)
   }
 
+  @Test
+  fun `setupLocationServices should fetch location when location is enabled and permissions granted`() {
+    val viewModel = mockk<QuestionnaireViewModel>()
+    val activity = spyk(QuestionnaireActivity())
+    val fusedLocationProviderClient = mockk<FusedLocationProviderClient>()
+    every { viewModel.applicationConfiguration.logQuestionnaireLocation } returns true
+    every { LocationServices.getFusedLocationProviderClient(activity) } returns
+      fusedLocationProviderClient
+    every { LocationUtils.isLocationEnabled(activity) } returns true
+    every { activity.hasLocationPermissions() } returns true
+    every { activity.fetchLocation(any()) } just runs
```

Comment:
Is it possible to test the actual functionality without using mocks? For example here `LocationUtils` is a utility class thus the return type for its functions will always be deterministic.